src/container/srv_target.c

```diff
@@ -2324,10 +2325,12 @@ cont_ec_eph_reduce(void *agg_arg, void *xs_arg)
 
 		c_eph = lookup_insert_cont_ec_eph(&pool->sp_ec_ephs_list,
 						  x_arg->ephs[i].cont_uuid);
-		if (c_eph->ce_eph == 0 ||
-		    (x_arg->ephs[i].eph != 0 &&
-		     x_arg->ephs[i].eph > c_eph->ce_last_eph))
+		if (c_eph->ce_first) {
 			c_eph->ce_eph = x_arg->ephs[i].eph;
+			c_eph->ce_first = 0;
+		} else if (x_arg->ephs[i].eph < c_eph->ce_eph) {
+			c_eph->ce_eph = x_arg->ephs[i].eph;
+		}
```

Comment:
Minor: The 'ce_first' looks bit confusing to me, why can't we just find out the smallest one and assign it to ce_eph?

BTW,  can we make sure it works well if any target/rank is in downout? (the eph on the target/rank never bumped)

Reply:
'ce_first' is for the case that originally the ce_eph is zero, and no one "x_arg->ephs[i].eph" can < it. So it need to assign one value for the first time.

For "can we make sure it works well if any target/rank is in downout? (the eph on the target/rank never bumped)" seems we need some similar logic as in "ds_cont_tgt_open(), ds_pool_get_failed_tgt_idx" for the collective. I can update in this patch if it need refresh later, or in a following patch.

Reply:
the suggested code seems not correct for some cases like:
eph array: [0]- 11, [1] 0, [2] 12
it will incorrectly set the result as 12

Reply:
hmm, it should be 0,   why 12?